### PR TITLE
WRR-1063: Add optional `options.preventScroll` parameter to `focus` function to prevent scroll by focus

### PR DESCRIPTION
**Pull request by 0x64**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
`Spotlight.focus` API does not support `preventScroll` option so that apps need to handle `focus` event separately when they need to handle focusing for its own UX.
In normal cases, the focus should be stay in visible area for users so that scrolling by focus is natural behavior. But if apps need to handle focus moving not for visual effect but for any other business logic, they may want to prevent scroll by focus. We'd like to support these cases.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
`Spotlight.focus` calls the standard `focus` API internally, so updated our API to pass the option to the standard API.
`Spotlight.focus` already has options called `containerOption` so updated this option parameter like below.
* Renamed `containerOption` to `options` in documents to avoid misunderstanding
* Added a new option `preventDefault`

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
This PR contains the original commit from #3177 and some more code refinements. Refer the original PR for the original contribution.

### Links
[//]: # (Related issues, references)
WRR-1063
#3177 (The original contribution)

### Comments
Enact-DCO-1.0-Signed-off-by: Nakjun Hwang (wns450@gmail.com)
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### CODE REVIEW

File: packages/spotlight/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ The following is a curated list of changes in the Enact spotlight module, newest
 
 ## [unreleased]
 
+### Added
```
Comment (MikyungKim) [inline packages/spotlight/CHANGELOG.md:7]: Let's sync up your branch with the lates develop.

File: packages/spotlight/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ The following is a curated list of changes in the Enact spotlight module, newest
...
+
+- `spotlight` an optional `options.preventScroll` parameter to `focus` function to prevent scrolling by focus
+
+### Changed
+
+- Renamed `spotlight` an optional `containerOption` parameter of `focus` function to `options` to avoid misunderstandings
```
[the remark this answers is not in the document]
Comment (0x64) [reply packages/spotlight/CHANGELOG.md:13]: I thought the same manner at first time but we already mentioned `containerOption` twice in changelogs. I wanted to remove this log honestly.

[the remark this answers is not in the document]
Comment (0x64) [reply packages/spotlight/CHANGELOG.md]: Well, removed. At least both of us do not want it. ;)

File: packages/spotlight/CHANGELOG.md (hunk 1 above)
Comment (0x64) [reply packages/spotlight/CHANGELOG.md:7]: done

**Review by MikyungKim — APPROVED** [review]
LGTM

**Review by juwonjeong — APPROVED** [review]
LGTM

**Pull request merged**